classification/keras/code/keras_script.py

```diff
@@ -0,0 +1,482 @@
+predictions = layers.Dense(1, activation="sigmoid", name="predictions")(x)
+
+keras_model = tf.keras.Model(inputs, predictions)
+
+# Compile the model with binary crossentropy loss and an adam optimizer.
+keras_model.compile(
+    loss=model_params["loss"], optimizer=model_params["optimizer"], metrics=model_params["metrics"]
+)
+
+
+# ### Train the model
+
+# (Neptune) Initialize the Neptune callback
+
+# In[56]:
+
+
+from neptune.new.integrations.tensorflow_keras import NeptuneCallback
+
+neptune_callback = NeptuneCallback(run=run, log_model_diagram=True, log_on_batch=True)
+
+
+# In[57]:
+
+
+training_params = {
+    "epochs": 3,
+}
+
+
+# In[58]:
+
+
+# Fit the model using the train and test datasets.
+keras_model.fit(
+    train_ds, validation_data=val_ds, epochs=training_params["epochs"], callbacks=neptune_callback
+)
+# Training parameters are logged automatically to Neptune
+
+
+# ### Evaluate the model
+
+# In[59]:
+
+
+_, acc = keras_model.evaluate(test_ds, callbacks=neptune_callback)
+
+
+# ## (Neptune) Associate run with model and vice-versa
+
+# In[60]:
+
+
+run_meta = {
+    "id": run.get_structure()["sys"]["id"].fetch(),
+    "name": run.get_structure()["sys"]["name"].fetch(),
+    "url": run.get_url(),
+}
+run_meta
+
+
+# In[61]:
+
+
+model_version["run"] = run_meta
+
+
+# In[62]:
+
+
+model_version_meta = {
+    "id": model_version.get_structure()["sys"]["id"].fetch(),
+    "name": model_version.get_structure()["sys"]["name"].fetch(),
+    "url": model_version.get_url(),
+}
+model_version_meta
+
+
+# In[63]:
+
+
+run["training/model/meta"] = model_version_meta
+
+
+# ## (Neptune) Upload serialized model and model weights to Neptune
+
+# In[64]:
+
+
+model_version["serialized_model"] = keras_model.to_json()
+
+
+# In[65]:
+
+
+keras_model.save_weights("model_weights.h5")
+model_version["model_weights"].upload("model_weights.h5")
+
+
+# (Neptune) Wait for all operations to sync with Neptune servers
+
+# In[66]:
+
+
+model_version.sync()
+
+
+# ## (Neptune) Promote best model to production
+
+# ### (Neptune) Fetch current production model
+
+# In[67]:
+
+
+model = neptune.init_model(with_id=f"{project_key}-KER")
+model_versions_df = model.fetch_model_versions_table().to_pandas()
+model.stop()
+model_versions_df
+
+
+# In[68]:
+
+
+production_models = model_versions_df[model_versions_df["sys/stage"] == "production"]["sys/id"]
+assert (
+    len(production_models) == 1
+), f"Multiple model versions found in production: {production_models.values}"
+
+
+# In[69]:
+
+
+prod_model_id = production_models.values[0]
+print(f"Current model in production: {prod_model_id}")
+
+
+# In[70]:
+
+
+prod_model = neptune.init_model_version(with_id=prod_model_id)
+prod_model_params = prod_model["params"].fetch()
+loaded_prod_model = tf.keras.models.model_from_json(
+    prod_model["serialized_model"].fetch(), custom_objects=None
+)
+
+prod_model["model_weights"].download()
+loaded_prod_model.load_weights("model_weights.h5")
```

Comment:
Would 
`prod_model` -> `npt_prod_model` and `loaded_prod_model` -> `prod_model` make more sense?